Sources/BottomSheet/BottomSheet.swift

```diff
@@ -7,55 +7,84 @@
-        self._isPresented = isPresented
-        self.sheetContent = sheetContent
-        self.view = view
+public struct BottomSheet<Content: View, SheetContent: View>: View {
+    @State var configuration = BottomSheetViewConfiguration()
+
+    private let isPresented: Bool
+    private let content: Content
+    private let sheetContent: SheetContent
+
+    public init(
+        isPresented: Bool = true,
+        @ViewBuilder content: () -> Content,
+        @ViewBuilder sheetContent: () -> SheetContent
+    ) {
+        self.isPresented = isPresented
+        self.content = content()
+        self.sheetContent = sheetContent()
     }
     
     public var body: some View {
         ZStack {
-            view
+            content
             
             if isPresented {
-                BottomSheetView(configuration: configuration, content: sheetContent)
-                    .ignoresSafeAreaEdges(configuration.ignoredEdges)
+                BottomSheetView(
+                    configuration: $configuration,
+                    content: sheetContent
+                )
             }
         }
     }
 }
 
 public extension View {
-    func bottomSheet<SheetContent:View>(
-        isPresented: Binding<Bool>,
-        sheetContent: @escaping ()
-        -> SheetContent) -> BottomSheet<SheetContent, Self> {
-            
-            BottomSheet(isPresented: isPresented, sheetContent: sheetContent, view: self)
-        }
+
+    func bottomSheet<SheetContent: View>(
+        isPresented: Bool = true,
+        @ViewBuilder sheetContent: () -> SheetContent
+    ) -> BottomSheet<Self, SheetContent> {
+        BottomSheet(
+            isPresented: isPresented,
+            content: { self },
+            sheetContent: sheetContent
+        )
+    }
+
 }
 
 private struct ExampleView: View {
+    let colors: [Color] = [.red, .orange, .yellow, .green, .blue, .indigo, .purple]
+
+    @ViewBuilder
+    var rainbowList: some View {
+        List {
+            ForEach((0..<colors.count * 4), id: \.self) { index in
+                let color = colors[index % colors.count]
+                Text(color.description)
+                    .foregroundColor(.white)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .listRowBackground(color)
+            }
+        }
+        .listStyle(.plain)
+    }
+
     var body: some View {
-        Text("Hello World!")
-            .bottomSheet(isPresented: .constant(true)) {
-                ScrollView {
-                    Text("Bottom Sheet")
-                        .padding(20)
-                }
+        TabView {
+            Tab("Map", systemImage: "map") {
+                Text("Map")
+                    .bottomSheet {
+                        rainbowList
+                    }
+                    .dragIndicatorPresentation(isVisible: true)
+                    .detentsPresentation(detents: [.small, .medium, .large])
```

Comment:
When `bottomSheet` is initially called, it hasn't received the `detentsPresentation(detents: [.small, .medium, .large])` modifier's changes. As a result, `detents` is initially `[.large]`, causing `selectedDetent` to always be `.large`, even if the developer doesn't desire `.large`. Working on resolving this issue now.

Reply:
Fixed in [8854da9](https://github.com/wojtek717/bottom-sheet/pull/8/commits/8854da952d5caf2960ceb7a4e24625c3a150071c). Added support for selecting the initial detent of the bottom sheet, and defaults to the smallest state.

https://github.com/user-attachments/assets/7293527a-a777-4e43-ac3d-1975eb14ecd1